main.tf

```diff
@@ -215,3 +215,23 @@ data "ibm_database_connection" "database_connection" {
   user_id       = ibm_database.elasticsearch.adminuser
   user_type     = "database"
 }
+
+resource "restapi_object" "put_trained_model" {
+  count         = var.enable_elser_model ? 1 : 0
+  path          = "//${data.ibm_database_connection.database_connection.https[0].hosts[0].hostname}:${data.ibm_database_connection.database_connection.https[0].hosts[0].port}/_ml/trained_models/.elser_model_1?pretty"
+  create_method = "PUT"
+  data = jsonencode({
+    input = {
+      field_names = ["text_field"]
+    }
+  })
+}
+
+resource "restapi_object" "start_trained_model_deployment" {
```

Comment:
It was my understanding that to interact with the database you need to use the database internal credentials not the IBM Cloud IAM credentials that seem to be used here. Not sure if this is related to the pipeline failure.